testing-tools/src/testnet_controllers/strom/node.rs

```diff
@@ -131,52 +156,86 @@ where
     where
         F: FnOnce(&StromNetworkManager<C>) -> R
     {
-        self.network.networks.strom_network(f)
+        self.state_lock.strom_network_manager(f)
     }
 
     pub fn strom_network_manager_mut<F, R>(&self, f: F) -> R
     where
         F: FnOnce(&mut StromNetworkManager<C>) -> R
     {
-        self.network.networks.strom_network_mut(f)
+        self.state_lock.strom_network_manager_mut(f)
     }
 
     pub fn eth_peer<F, R>(&self, f: F) -> R
     where
         F: FnOnce(&Peer<C>) -> R
     {
-        self.network.networks.eth_peer(f)
+        self.state_lock.eth_peer(f)
     }
 
     pub fn eth_peer_mut<F, R>(&self, f: F) -> R
     where
         F: FnOnce(&mut Peer<C>) -> R
     {
-        self.network.networks.eth_peer_mut(f)
+        self.state_lock.eth_peer_mut(f)
     }
 
-    pub fn start_network(&self, blocking: bool) {
-        if blocking {
-            self.network.blocking_start_network()
-        } else {
-            self.network.start_network()
-        }
+    pub fn start_network(&self) {
+        self.state_lock.set_network(true);
     }
 
-    pub fn stop_network(&self, blocking: bool) {
-        if blocking {
-            self.network.blocking_stop_network()
-        } else {
-            self.network.stop_network()
-        }
+    pub fn stop_network(&self) {
+        self.state_lock.set_network(false);
     }
 
     pub fn is_network_on(&self) -> bool {
-        self.network.is_network_on()
+        self.state_lock.network_state()
     }
 
     pub fn is_network_off(&self) -> bool {
-        self.network.is_network_off()
+        !self.state_lock.network_state()
+    }
+
+    pub fn start_network_and_consensus(&self) {
+        self.start_network();
+        self.start_conensus();
+    }
+
+    pub fn stop_network_and_consensus(&self) {
+        self.stop_network();
+        self.stop_consensus();
+    }
+
+    /// Consensus
+    /// -------------------------------------
```

Comment:
```suggestion
    // Consensus
    // -------------------------------------
    
```
don't make it a doc comment on the function. just do as a reg comment